### PR TITLE
add haptics support :)

### PR DESCRIPTION
**Pull request by ethernet8023**
adds the ability to mess with a Macbook touchpad using `NSHapticFeedbackManager`.

### CODE REVIEW

File: src/appkit/haptics.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+use std::convert::TryFrom;
+
+use objc::{class, msg_send, runtime::Object, sel, sel_impl};
+use objc_id::ShareId;
+
+use crate::foundation::NSUInteger;
+
+#[derive(Clone, Debug)]
+pub struct HapticFeedbackPerformer(pub ShareId<Object>);
+impl HapticFeedbackPerformer {
```
Comment (ryanmcgrath) [inline src/appkit/haptics.rs:11]: This is so dumb-pedantic of me, but can you linebreak between the struct and impl?

**Comment by ryanmcgrath** [comment]
This is cool and I'm down to merge it as-is - the iOS tests that are failing are due to something else as far as I know, and I need to fix them before 0.4 gets released... so we can ignore it for now.

If you hit that linebreak that I mentioned I'll hit that merge button, lol

File: src/appkit/haptics.rs (hunk 1 above)
Comment (ethernet8023) [reply src/appkit/haptics.rs:11]: LMAO no problem

**Comment by ethernet8023** [comment]
comments addressed - also check this out, there's internal APIs that make it vibrate harder
https://gist.github.com/arilotter/9e273bed948c6609fe17e429a54c3e8d

**Comment by ryanmcgrath** [comment]
> comments addressed - also check this out, there's internal APIs that make it vibrate harder
> https://gist.github.com/arilotter/9e273bed948c6609fe17e429a54c3e8d

Yup, those APIs often feel like they go unnoticed but they're pretty cool.

There's actually a world where this might be better off as an external library, but I'm fine with it in cacao for now since these things are generally so tied to UI to begin with.

Thanks for the contribution!

**Pull request merged**